tensorflow/core/kernels/cwise_op_abs.cc

```diff
@@ -38,7 +38,11 @@ REGISTER_KERNEL_BUILDER(Name("Abs")
 #endif
 
 #if TENSORFLOW_USE_SYCL
-REGISTER3(UnaryOp, SYCL, "Abs", functor::abs, float, double, int64);
+#define REGISTER_KERNEL(type) REGISTER(UnaryOp, SYCL, "Abs", functor::abs, type);
```

Comment:
i tried it, but did not found a way to implement it. My code looks like below, while it is illegal grammar.  Do you have any idea @benoitsteiner , thanks.

#define REGISTER_NUMBER_SYCL(OP, N, F)          \
  #define REGISTER_SYCL_KERNEL(type) REGISTER(OP, SYCL, N, F, type)   \
  TF_CALL_SYCL_NUMBER_TYPES(REGISTER_SYCL_KERNEL);                    \
  #undef REGISTER_SYCL_KERNEL